app.rb

```diff
@@ -57,54 +74,187 @@ def create_book
     book = Book.new(title, author)
     @books.push(book)
     puts "Book '#{title}' created successfully"
+    save_books_to_json
   end
 
   def create_rental
     puts 'Select a book from the following list by number'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
-    end
-    book_index = gets.chomp.to_i
+    list_books
+    book_index = gets.chomp.to_i - 1
+
     puts 'Select a person from the following list by number (not id)'
-    @people.each_with_index do |person, index|
-      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
-      puts "#{index}) [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
-    person_index = gets.chomp.to_i
+    list_people
+    person_index = gets.chomp.to_i - 1
+
     print 'Date: '
     date = gets.chomp
-    @rentals.push(Rental.new(date, @books[book_index], @people[person_index]))
+
+    rental = Rental.new(date, @books[book_index], @people[person_index])
+    @rentals.push(rental)
     puts 'Rental created successfully'
+    save_rentals_to_json
   end
 
+  #------------- lists
+
   def list_books
+    puts 'List of Books:'
     @books.each_with_index do |book, index|
-      puts "#{index} - Title: #{book.title}, Author: #{book.author}"
+      puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
     end
   end
 
   def list_people
     @people.each_with_index do |person, index|
-      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
-      puts "#{index} - [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      if person.instance_of?(Student)
+        type = 'Student'
+      elsif person.instance_of?(Teacher)
+        type = 'Teacher'
+      end
+
+      puts "#{index + 1} - [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
   end

```

Comment:
- [ ] After creating a book, person and rental and exiting the app on restart of the files shown in the image below, there occurs an error with the people.json file and as such when I try to read the data for people, there is no data showing, please kindly fix this issue as it is also affecting the rentals because right now I cannot see the rental for the person which I created 👍 
![ppl](https://github.com/rodrigopenayo98/OOP-school-librery/assets/89055502/7aa7e858-45c1-42a4-98e4-9f635ac9e63d)